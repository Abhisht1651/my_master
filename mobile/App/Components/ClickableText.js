// @flow
import React from 'react'
import { Text, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

type Props = {
  onPress?: PropTypes.func,
  textStyle?: StyleSheet,
  children?: string,
  style?: StyleSheet,
  text?: string
}
const ClickableText = (props: Props) => {
  const { style, onPress, textStyle, children, text } = props
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      <Text style={textStyle}> {children || text} </Text>
    </TouchableOpacity>
  )
}

ClickableText.defaultProps = {
  onPress: () => {},
  textStyle: StyleSheet.create({}),
  children: '',
  style: StyleSheet.create({}),
  text: ''
}

export default ClickableText
