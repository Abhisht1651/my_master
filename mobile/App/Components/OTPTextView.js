// https://github.com/naveenvignesh5/react-native-otp-textinput/blob/master/index.js
// @flow
import React, { useState } from 'react'
import { View, TextInput, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  textInput: {
    textAlign: 'center',
    fontSize: 22
  }
})

const getOTPTextChucks = (inputCount, inputCellLength, text) => {
  let otpText = text.match(new RegExp(`.{1,${inputCellLength}}`, 'g')) || []

  otpText = otpText.slice(0, inputCount)

  return otpText
}

type Props = {
  defaultValue: PropTypes.String,
  inputCount: PropTypes.Sumber,
  containerStyle: StyleSheet,
  textInputStyle: StyleSheet,
  inputCellLength: PropTypes.Number,
  tintColor: PropTypes.String,
  offTintStyle: StyleSheet,
  handleTextChange: PropTypes.Func,
  keyboardType: PropTypes.String,
  changedTextBorder: StyleSheet
}
const OTPTextView = (props: Props) => {
  const {
    inputCount,
    offTintStyle,
    tintColor,
    inputCellLength,
    containerStyle,
    textInputStyle,
    keyboardType,
    changedTextBorder,
    defaultValue,
    handleTextChange
  } = props
  const [focusedInput, setfocusedInput] = useState(0)
  const [otpText, setotpText] = useState(
    getOTPTextChucks(inputCount, inputCellLength, defaultValue)
  )

  const [inputs, setinputs] = useState([])

  const basicValidation = (text) => {
    const validText = /^[0-9a-zA-Z]+$/
    return text.match(validText)
  }

  const onTextChange = (text, i) => {
    if (text && !basicValidation(text)) {
      return
    }
    const oll = otpText
    oll[i] = text
    setotpText(oll)
    handleTextChange(otpText.join(''))
    if (text.length === inputCellLength && i !== inputCount - 1) {
      // $FlowFixMe
      inputs[i + 1].focus()
    }
  }

  const onInputFocus = (i) => {
    const prevIndex = i - 1

    if (prevIndex > -1 && !otpText[prevIndex] && !otpText.join('')) {
      // $FlowFixMe
      inputs[prevIndex].focus()
      return
    }

    setfocusedInput(i)
  }

  const onKeyPress = (e, i) => {
    const val = otpText[i] || ''

    if (e.nativeEvent.key === 'Backspace' && i !== 0 && (!(val.length - 1) || val.length === 0)) {
      // $FlowFixMe
      inputs[i - 1].focus()
    }
  }

  // const clear = () => {
  //   setotpText([])
  //   inputs[0].focus()
  // }

  // const setValue = (value) => {
  //   setotpText(getOTPTextChucks(inputCount, inputCellLength, value))
  //   handleTextChange(value)
  // }

  const TextInputs = []

  for (let i = 0; i < inputCount; i += 1) {
    const inputStyle = [styles.textInput, textInputStyle, offTintStyle]

    if (focusedInput === i && tintColor) {
      inputStyle.push({ borderColor: tintColor })
    }
    if (otpText[i]) {
      inputStyle.push(changedTextBorder)
    }

    TextInputs.push(
      <TextInput
        ref={(e) => {
          const x = inputs
          x[i] = e
          setinputs(x)
        }}
        key={i}
        autoCorrect={false}
        keyboardType={keyboardType}
        autoFocus={false}
        value={otpText[i] || ''}
        style={inputStyle}
        maxLength={inputCellLength}
        onFocus={() => onInputFocus(i)}
        onChangeText={(text) => onTextChange(text, i)}
        multiline={false}
        onKeyPress={(e) => onKeyPress(e, i)}
      />
    )
  }

  return <View style={[styles.container, containerStyle]}>{TextInputs}</View>
}

OTPTextView.defaultProps = {
  defaultValue: '',
  inputCount: 4,
  tintColor: '',
  offTintStyle: {},
  inputCellLength: 1,
  containerStyle: {},
  textInputStyle: {},
  handleTextChange: () => {},
  keyboardType: 'numeric',
  changedTextBorder: {}
}

export default OTPTextView
