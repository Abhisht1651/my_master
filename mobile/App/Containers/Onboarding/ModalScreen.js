// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React, { useState, useEffect } from 'react'
import {
  Text,
  Image,
  View,
  TouchableOpacity,
  KeyboardAvoidingView,
  Platform,
  Keyboard
} from 'react-native'
import OTPTextView from '../../Components/OTPTextView'
import ClickableText from '../../Components/ClickableText'
import { MediumThinButton } from '../../Components/Buttons/MediumThinButton'
import styles from '../Styles/Onboarding/ModalScreenStyle'
import Images from '../../Themes/Images'
// Styles
type Props = NavigationContainerProps & {}

const ForgotPasswordOTP = (props: Props) => {
  const [otpInput, setotpInput] = useState('')
  const [seconds, setSeconds] = useState(60)
  const [isActive, setIsActive] = useState(true)
  const [isKeyboardVisible, setKeyboardVisible] = useState(false)
  useEffect(() => {
    let interval = null
    if (seconds === 0) {
      setIsActive(false)
    }
    if (isActive) {
      interval = setInterval(() => {
        setSeconds(seconds - 1)
      }, 1000)
    } else if (!isActive && seconds !== 60) {
      clearInterval(interval)
    }
    return () => clearInterval(interval)
  }, [isActive, seconds])

  useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', () => {
      setKeyboardVisible(true) // or some other action
    })
    const keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', () => {
      setKeyboardVisible(false) // or some other action
    })
    return () => {
      keyboardDidHideListener.remove()
      keyboardDidShowListener.remove()
    }
  }, [])
  const button = (prop: Props) => {
    const len = otpInput.length
    return (
      <View style={isKeyboardVisible ? styles.modalButtonK : styles.modalButtonNK}>
        <MediumThinButton
          title="Continue"
          disabled={len !== 6}
          onPress={len === 6 ? () => prop.navigation.navigate('SignUpDetails') : null}
        />
      </View>
    )
  }
  return (
    <KeyboardAvoidingView
      style={styles.centeredView}
      behavior={Platform.OS === 'ios' ? 'padding' : 'padding'}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <View style={styles.header}>
        <TouchableOpacity
          onPress={() => props.navigation.goBack()}
          style={styles.backArrowContainer}>
          <Image source={Images.leftArrow} style={styles.backArrow} />
        </TouchableOpacity>
        <Text style={styles.modalHeading}>Verification Code</Text>
        <TouchableOpacity onPress={() => props.navigation.goBack()}>
          <View>
            <Image source={Images.closeIconPurple} style={styles.cross} />
          </View>
        </TouchableOpacity>
      </View>

      <Text style={isKeyboardVisible ? styles.modalTextK : styles.modalText}>
        Please add verification code sent to your email.
      </Text>
      <OTPTextView
        containerStyle={styles.rowView}
        handleTextChange={(text) => setotpInput(text)}
        inputCount={6}
        keyboardType="numeric"
        textInputStyle={styles.inputText}
        changedTextBorder={{ borderBottomWidth: 1, borderColor: '#1F1F60' }}
        offTintStyle={styles.offTint}
      />

      <View style={styles.resendContainer}>
        <View style={styles.rowView}>
          <Text style={styles.coden}>Havenâ€™t received code?</Text>
          <ClickableText textStyle={styles.otpresend} text="Resend" />
        </View>

        <Text style={[styles.otpresend, styles.otpTimer]}>00:{seconds} sec</Text>
      </View>
      {button(props)}
    </KeyboardAvoidingView>
  )
}
export default ForgotPasswordOTP
