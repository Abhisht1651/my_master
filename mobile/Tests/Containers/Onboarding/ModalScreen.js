import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ModalScreen from '../../../App/Containers/Onboarding/ModalScreen'

import OTPTextView from '../../../App/Components/OTPTextView'
import ClickableText from '../../../App/Components/ClickableText'
import { MediumThinButton } from '../../../App/Components/Buttons/MediumThinButton'

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

test('ModalScreen component renders correctly', () => {
  const props = createTestProps({})
  const tree = renderer.create(<ModalScreen {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe('ModalScreen ', () => {
  describe('rendering', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<ModalScreen {...props} />) // no compile-time error
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('There should be 1 ClickableText', () => {
      expect(wrapper.find(ClickableText)).toHaveLength(1)
    })
    it('There should be 1 OTP', () => {
      expect(wrapper.find(OTPTextView)).toHaveLength(1)
    })
    it('Match number of OTP text', () => {
      const grad = wrapper.find(OTPTextView).at(0)
      expect(grad.prop('inputCount')).toBe(6)
    })
    it('There should be 1 RoundedButton', () => {
      expect(wrapper.find(MediumThinButton)).toHaveLength(1)
    })
  })
})
