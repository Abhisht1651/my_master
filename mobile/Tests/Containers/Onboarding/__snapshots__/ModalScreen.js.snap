// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalScreen  rendering should match to snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 100.74479166666667,
        "justifyContent": "space-between",
        "paddingHorizontal": 64,
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 37,
          "letterSpacing": -0.14,
          "textAlign": "center",
        }
      }
    >
      Verification Code
    </Text>
    <ForwardRef
      onPress={[Function]}
    >
      <View>
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 61.497400000000006,
              "resizeMode": "contain",
              "tintColor": "#7B46E4",
              "width": 34.575,
            }
          }
        />
      </View>
    </ForwardRef>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Light",
        "fontSize": 30,
        "letterSpacing": -0.11,
        "marginBottom": 44.29296875,
        "marginHorizontal": 114,
        "marginTop": 138.95833333333334,
        "textAlign": "center",
      }
    }
  >
    Please add verification code sent to your email.
  </Text>
  <OTPTextView
    changedTextBorder={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#1F1F60",
      }
    }
    containerStyle={
      Object {
        "flexDirection": "row",
      }
    }
    defaultValue=""
    handleTextChange={[Function]}
    inputCellLength={1}
    inputCount={6}
    keyboardType="numeric"
    offTintStyle={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#DCDCE9",
      }
    }
    textInputStyle={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Bold",
        "fontSize": 61,
        "height": 93.38000000000001,
        "margin": 0,
        "marginLeft": 20,
        "padding": 0,
        "textAlign": "center",
        "width": 68,
      }
    }
    tintColor=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 78.1640625,
        "marginTop": 41.6875,
        "width": 448,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
          }
        }
      >
        Havenâ€™t received code?
      </Text>
      <ClickableText
        onPress={[Function]}
        style={Object {}}
        text="Resend"
        textStyle={
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 28,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 28,
          },
          Object {
            "color": "#1F1F60",
            "marginTop": 8.684895833333334,
          },
        ]
      }
    >
      00:
      60
       sec
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 521.09375,
      }
    }
  >
    <MediumThinButton
      disabled={true}
      onPress={null}
      style={Object {}}
      title="Continue"
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`ModalScreen component renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 100.74479166666667,
        "justifyContent": "space-between",
        "paddingHorizontal": 64,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 37,
          "letterSpacing": -0.14,
          "textAlign": "center",
        }
      }
    >
      Verification Code
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 61.497400000000006,
              "resizeMode": "contain",
              "tintColor": "#7B46E4",
              "width": 34.575,
            }
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Light",
        "fontSize": 30,
        "letterSpacing": -0.11,
        "marginBottom": 44.29296875,
        "marginHorizontal": 114,
        "marginTop": 138.95833333333334,
        "textAlign": "center",
      }
    }
  >
    Please add verification code sent to your email.
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={false}
      keyboardType="numeric"
      maxLength={1}
      multiline={false}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 22,
            "textAlign": "center",
          },
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 61,
            "height": 93.38000000000001,
            "margin": 0,
            "marginLeft": 20,
            "padding": 0,
            "textAlign": "center",
            "width": 68,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#DCDCE9",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 78.1640625,
        "marginTop": 41.6875,
        "width": 448,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
          }
        }
      >
        Havenâ€™t received code?
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 28,
            }
          }
        >
           
          Resend
           
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 28,
          },
          Object {
            "color": "#1F1F60",
            "marginTop": 8.684895833333334,
          },
        ]
      }
    >
      00:
      60
       sec
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 521.09375,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#7B46E466",
            "borderRadius": 78,
            "height": 100,
            "width": 550,
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "BananaGrotesk-Bold",
                "fontSize": 32,
                "letterSpacing": -0.18,
                "lineHeight": 36,
              },
              Object {},
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;
